cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs

```diff
@@ -1161,28 +1169,86 @@ instance
+        object
+          [ "kind" .= String "CollectError"
+          , "error" .= String "NoCostModel"
+          , "language" .= toJSON lang
+          ]
+      Alonzo.BadTranslation err ->
+        object
+          [ "kind" .= String "PlutusTranslationError"
+          , "error" .= case err of
+              Babbage.AlonzoContextError _alonzoCtxErr ->
+                String "???"
+              Babbage.ByronTxOutInContext txOutSource ->
                 String $
                   "Cannot construct a Plutus ScriptContext from this transaction "
                     <> "due to a Byron UTxO being created or spent: "
                     <> textShow txOutSource
-              Alonzo.TranslationLogicMissingInput txin ->
-                String $ "Transaction input does not exist in the UTxO: " <> textShow txin
-              Alonzo.RdmrPtrPointsToNothing ptr ->
+              Babbage.RdmrPtrPointsToNothing ptr ->
                 object
                   [ "kind" .= String "RedeemerPointerPointsToNothing"
                   , "ptr" .= Api.fromAlonzoRdmrPtr ptr
                   ]
-              Alonzo.LanguageNotSupported lang ->
-                String $ "Language not supported: " <> textShow lang
-              Alonzo.InlineDatumsNotSupported txOutSource ->
+              Babbage.InlineDatumsNotSupported txOutSource ->
                 String $ "Inline datums not supported, output source: " <> textShow txOutSource
-              Alonzo.ReferenceScriptsNotSupported txOutSource ->
+              Babbage.ReferenceScriptsNotSupported txOutSource ->
                 String $ "Reference scripts not supported, output source: " <> textShow txOutSource
-              Alonzo.ReferenceInputsNotSupported txins ->
+              Babbage.ReferenceInputsNotSupported txins ->
                 String $ "Reference inputs not supported: " <> textShow txins
-              Alonzo.TimeTranslationPastHorizon msg ->
-                String $ "Time translation requested past the horizon: " <> textShow msg
+          ]
+
+instance ToJSON (Alonzo.CollectError (Conway.ConwayEra StandardCrypto)) where
+  toJSON cError =
+    case cError of
+      Alonzo.NoRedeemer sPurpose ->
+        object
+          [ "kind" .= String "CollectError"
+          , "error" .= String "NoRedeemer"
+          , "scriptpurpose" .= renderScriptPurpose sPurpose
+          ]
+      Alonzo.NoWitness sHash ->
+        object
+          [ "kind" .= String "CollectError"
+          , "error" .= String "NoWitness"
+          , "scripthash" .= renderScriptHash sHash
+          ]
+      Alonzo.NoCostModel lang ->
+        object
+          [ "kind" .= String "CollectError"
+          , "error" .= String "NoCostModel"
+          , "language" .= toJSON lang
+          ]
+      Alonzo.BadTranslation err ->
+        object
+          [ "kind" .= String "PlutusTranslationError"
+          , "error" .= case err of
+              Conway.BabbageContextError _babbageCtxErr ->
+                String "???"
```

Comment:
@lehins

Reply:
Use instances in this PR instead for `CollectEror`